### PR TITLE
ISSUE-80: Add patch support for psycopg2 

### PR DESCRIPTION
**Pull request by Tankanow**
Issue 80

Add patch support for [psycopg2](https://pypi.org/project/psycopg2) using existing `dbapi2` tracing classes. For now, add `host`, `database`, `user`, and `database_type` to `sql` metadata.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_xray_sdk/ext/psycopg2/patch.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        'connect',
+        _xray_traced_connect
+    )
+
+
+def _xray_traced_connect(wrapped, instance, args, kwargs):
+
+    conn = wrapped(*args, **kwargs)
+    meta = {
+        'database_type': 'postgresql',
```
Comment (Tankanow) [inline aws_xray_sdk/ext/psycopg2/patch.py]: @haotianw465, how do I see the custom metadata in XRay? I system tested this code by deploying it along with one of my Lambda functions; I did see the correct `database_type`, however I did not see `database_host`, `database_name`, or `database_user` in the Raw Trace Data in the AWS XRay Console.

Comment (haotianw465) [reply aws_xray_sdk/ext/psycopg2/patch.py]: The service will drop unknown keys. Here is a sample sql document the service support.

```json

"sql" : {
     "url": "jdbc:postgresql://aawijb5u25wdoy.cpamxznpdoq8.us-west-2.rds.amazonaws.com:5432/ebdb",
     "preparation": "statement",
     "database_type": "PostgreSQL",
     "database_version": "9.5.4",
     "driver_version": "PostgreSQL 9.4.1211.jre7",
     "user" : "dbuser"
}
```

**Comment by Tankanow** [comment]
@haotianw465, I updated the metadata and confirmed via system test that it shows up in the AWS XRay Console. Please let me know if there is anything else I should do for this PR.

**Comment by haotianw465** [comment]
Thank you for your contribution. I have merged your PR and will roll it out in our next release.

**Pull request merged**